src/executor/mod.rs

```diff
@@ -163,7 +169,13 @@ impl<D: Deliverable> Future for Executor<D> {
                                     self.transaction_timeout.clone(),
                                     counter
                                 );
-                            },
+
+                                self.transactions_processed += 1;
+                                if self.transactions_processed > self.max_total_transactions_per_worker {
+                                    state_changed = true;
+                                    break ExecutorState::Draining;
+                                }
+                            }
```

Comment:
Would it be reasonable to prevent queuing up transactions in excess of the remaining amount that will be processed?

Reply:
discussed offline, lgtm

Reply:
It should be okay as we want this client to stop processing transactions after the limit, dropped transactions will be retried with another client.

Reply:
How does this work with in-flight transactions?

Reply:
In-flight transactions are still tracked and the executor is held in the Draining state until they are finished. However, the receiver is dropped the moment it transitions to the Draining state, so from the pools point of view the executor is gone and will be re-created.